src/adapter.js

```diff
@@ -355,10 +361,29 @@ function createStartFn (karma, jasmineEnv) {
 
     jasmineEnv = jasmineEnv || window.jasmine.getEnv()
 
-    setOption(jasmineConfig.stopOnFailure, jasmineEnv.throwOnExpectationFailure)
-    setOption(jasmineConfig.failFast, jasmineEnv.stopOnSpecFailure)
-    setOption(jasmineConfig.seed, jasmineEnv.seed)
-    setOption(jasmineConfig.random, jasmineEnv.randomizeTests)
+    if (typeof jasmineEnv.configure === 'function') {
+      // it is actually possible to pass jasmineConfig directly to `jasmine.configure`,
+      // but that would break karma-jasmine's documented use of `stopOnFailure`
+      // which is not a valid jasmine config option
+
+      if (jasmineConfig.hasOwnProperty('stopOnFailure')) {
+        jasmineEnv.configure({ oneFailurePerSpec: jasmineConfig.stopOnFailure })
+      }
+      if (jasmineConfig.hasOwnProperty('failFast')) {
+        jasmineEnv.configure({ failFast: jasmineConfig.failFast })
+      }
+      if (jasmineConfig.hasOwnProperty('seed')) {
+        jasmineEnv.configure({ seed: jasmineConfig.seed })
+      }
+      if (jasmineConfig.hasOwnProperty('random')) {
+        jasmineEnv.configure({ random: jasmineConfig.random })
+      }
+    } else {
+      setOption(jasmineConfig.stopOnFailure, jasmineEnv.throwOnExpectationFailure)
+      setOption(jasmineConfig.failFast, jasmineEnv.stopOnSpecFailure)
```

Comment:
Why aren't these values consistent with the ones above?

Reply:
These values are just indented to fit within the else block. They are otherwise unchanged from the previous version of `karma-jasmine`